### PR TITLE
Fix singularity image

### PR DESCRIPTION
**Pull request by NucciTheBoss**
### Purpose

Pull request fixes tzdata data-entry issue brought up in issue #4365. 

### Changelog

1. Set timezone to Etc/UTC to make tzdata data-entry prompt go away.
2. Updated base image to `nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04`. `g++` was too old to compile `numpy` in the old base image.
3. Removed `make` in `apt-get install -y --no-install-recommends`. `make` is provided by the `build-essential` package.
4. Added missing dependencies to `apt-get install -y --no-install-recommends`. Needed `gfortran`, `liblapack-dev`, `libopenblas-dev`, and `libhdf5-dev`.
5. Added some code styling to build recipe.

### Testing

Had some of the consultants I work with run tests on our cluster. Did not encounter any runtime errors.

### What's next

Updating `Dockerfile` and `singularity` recipe to newer version of pytorch_geometric.

### CODE REVIEW

File: docker/singularity [hunk 1]
```diff
@@ -1,53 +1,73 @@
...
+    libopenblas-dev \
+    libhdf5-dev
 
-export PYENV_ROOT=/opt/pyenv
-export PATH="/opt/pyenv/bin:$PATH"
-curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-pyenv install 3.7.2
-echo 'export PATH=/opt/pyenv/versions/3.7.2/bin/:$PATH' >> $SINGULARITY_ENVIRONMENT
-export PATH=/opt/pyenv/versions/3.7.2/bin/:$PATH
+  export PYENV_ROOT=/opt/pyenv
```
Comment (Padarn) [inline docker/singularity:34]: Any reason we have added spaces infront of all of these?

Comment (NucciTheBoss) [reply docker/singularity:34]: Spaces were added for readability/styling. Styled the recipe similar to the Sylabs [example recipe](https://sylabs.io/guides/3.8/user-guide/definition_files.html).

If its preferred to keep the recipe the original way, say the word and I can revert it back.

Comment (Padarn) [reply docker/singularity:34]: No preference from me, I was just curious :-)

**Review by rusty1s — APPROVED** [review]
LGTM. We could upgrade to a reasonable PyTorch version in a later PR.

File: docker/singularity [hunk 2]
```diff
@@ -1,53 +1,73 @@
...
+From: nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04
 
 %post
-CURDIR=`pwd`
-
-apt-get update
-apt-get install -y tmux nano git wget
-
-apt-get install -y --no-install-recommends make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+  CURDIR=$(pwd)
```
[the remark this answers is not in the document]
Comment (NucciTheBoss) [reply docker/singularity:5]: From how it is in the recipe, `CURDIR` is used snapshot the current working directory file path before changing into the `sing_build_cuda` directory. `CURDIR` is then used again when we go to clean up the image before producing the container. I kept this from the earlier recipe, however, we might be able to remove it.

I can comment in the recipe why `CURDIR` is here now, and then we could see if we can remove it in the next pull request?

[the remark this answers is not in the document]
Comment (rusty1s) [reply docker/singularity:5]: Sorry for the confusion. I was actually referring to the line below, the `TZ` line (the reason of this PR).

[the remark this answers is not in the document]
Comment (NucciTheBoss) [reply docker/singularity:5]: My apologies! Yes I can add a comment. How does the following comment beforehand sound?

```
# Set timezone to Etc/UTC for tzdata. See issue #4365 for more details.
TZ=Etc/UTC && \
    ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && \
    echo $TZ > /etc/timezone
```

[the remark this answers is not in the document]
Comment (rusty1s) [reply docker/singularity:5]: Perfect :)

**Comment by NucciTheBoss** [comment]
Added explanation comment for setting timezone to recipe. Should be good to go.

**Comment by rusty1s** [comment]
Thank you!

**Pull request merged**